components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/LoginEventHookHandler.java

```diff
@@ -0,0 +1,230 @@
+                IdentityEventConstants.EventName.AUTHENTICATION_FAILURE.name().equals(eventName);
+    }
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        EventData eventData = eventHookHandlerUtils.buildEventDataProvider(event);
+
+        if (eventData.getAuthenticationContext().isPassiveAuthenticate()) {
+            return;
+        }
+
+        //TODO: Add the implementation to read the Event Schema Type from the Tenant Configuration
+        LoginEventPayloadBuilder payloadBuilder = PayloadBuilderFactory
+                .getLoginEventPayloadBuilder(EVENT_SCHEMA_TYPE_WSO2);
+        EventPublisherConfig loginEventPublisherConfig = null;
+        try {
+            loginEventPublisherConfig = getLoginEventPublisherConfigForTenant(
+                    eventData.getAuthenticationContext().getLoginTenantDomain(), event.getEventName());
+
+            EventPayload eventPayload;
+            String eventUri;
+
+            if (IdentityEventConstants.EventName.AUTHENTICATION_SUCCESS.name().equals(event.getEventName()) &&
+                    loginEventPublisherConfig.isPublishEnabled()) {
+                eventPayload = payloadBuilder.buildAuthenticationSuccessEvent(eventData);
+                eventUri = eventConfigManager.getEventUri(Constants.EventHandlerKey.LOGIN_SUCCESS_EVENT);
+                String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+                SecurityEventTokenPayload securityEventTokenPayload = eventHookHandlerUtils.buildSecurityEventToken(eventPayload, eventUri);
+                eventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+            } else if (IdentityEventConstants.EventName.AUTHENTICATION_STEP_FAILURE.name().equals(event.getEventName()) &&
+                    loginEventPublisherConfig.isPublishEnabled()) {
+                eventPayload = payloadBuilder.buildAuthenticationFailedEvent(eventData);
+                eventUri = eventConfigManager.getEventUri(Constants.EventHandlerKey.LOGIN_FAILED_EVENT);
+                String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+                SecurityEventTokenPayload securityEventTokenPayload = eventHookHandlerUtils.buildSecurityEventToken(eventPayload, eventUri);
+                eventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+            }
+        } catch (IdentityEventException e) {
+            log.error("Error while retrieving event publisher configuration for tenant.", e);
+        }
+    }
+
+    private EventPublisherConfig getLoginEventPublisherConfigForTenant(String tenantDomain, String eventName)
+            throws IdentityEventException {
+
+        if (StringUtils.isEmpty(tenantDomain) || MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            throw new IdentityEventException("Invalid tenant domain: " + tenantDomain);
```

Comment:
Why we throw this if tenantdomain is equal to the super tenant ?

Reply:
added todo for this https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/248a1a04b34ee30bc9b512092e84c0397489ea35